messaging/usecase/room_usecase.go

```diff
@@ -0,0 +1,29 @@
+package usecase
+
+import (
+	"chat/domain"
+	"chat/utils/errors"
+	"context"
+	_ "reflect"
+	"time"
+)
+type roomUseCase struct {
+	rr      domain.RoomRepository
+	timeout time.Duration
+}
+
+func (u *roomUseCase) SaveRoom(ctx context.Context, room *domain.ChatRoom) error {
+	ctx, cancel := context.WithTimeout(ctx, u.timeout)
+	defer cancel()
+	err := u.rr.SaveRoom( room )
+	if err!=nil {
+		return err
+	}
+
+	anyExistingRoom, err := u.rr.GetRoom( room.RoomID )
```

Comment:
shouldn't we check if the room exists before trying to add it? That way we don't try to add it unnecessarily